src/main/java/com/limechain/trie/structure/TrieStructure.java

```diff
@@ -91,6 +91,31 @@ public Entry<T> node(Nibbles key) {
         };
     }
 
+    /**
+     * Insert a branch node at the given key.
+     *
+     * @param key      partial key in nibbles
+     * @param nodeData data to insert
+     * @implSpec if a storage value is already present for the given key, it will be overwritten
+     */
+    public void insertBranch(Nibbles key, T nodeData) {
```

Comment:
Let's add a comment above this public method that it's a temporary hacky solution in order to properly load the trie structure from the DB... and not a very polished part of the public interface of the class.